tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+      const relativePath = `undefined/undefined`; // its undefined because 'findRecord' function is executes when job type is "Update" and return record as undefined while testing
+      expect(createMergeTilesTasksMock).toHaveBeenCalledWith(
+        testData,
+        relativePath,
+        TaskAction.MERGE_TILES,
+        JobAction.UPDATE,
+        [Grid.TWO_ON_ONE],
+        [0, 0, 1, 1],
+        managerCallbackUrl
+      );
+    });
+
+    it('should throw Bad Request Error for "Update" job type if layer is not exists in map proxy', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+      getHighestLayerVersionMock.mockResolvedValue([1.0, 2.0]);
+      mapExistsMock.mockResolvedValue(false);
+      findJobsMock.mockResolvedValue([]);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      validateGpkgFilesMock.mockReturnValue(true);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createMergeTilesTasksMock.mockResolvedValue(undefined);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+
+      await expect(action).rejects.toThrow(BadRequestError);
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(1);
+      expect(fileValidatorValidateExistsMock).toHaveBeenCalledTimes(1);
+      expect(findJobsMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should throw Bad Request Error for "New" or "Update" job type if higher product version is already exists in catalog', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(4.0);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy